pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/LLCSegmentCommit.java

```diff
@@ -0,0 +1,152 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.controller.api.restlet.resources;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.apache.commons.io.FileUtils;
+import org.restlet.ext.fileupload.RestletFileUpload;
+import org.restlet.representation.Representation;
+import org.restlet.representation.StringRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.linkedin.pinot.common.config.TableNameBuilder;
+import com.linkedin.pinot.common.protocols.SegmentCompletionProtocol;
+import com.linkedin.pinot.common.restlet.swagger.Description;
+import com.linkedin.pinot.common.restlet.swagger.HttpVerb;
+import com.linkedin.pinot.common.restlet.swagger.Paths;
+import com.linkedin.pinot.common.restlet.swagger.Summary;
+import com.linkedin.pinot.common.utils.LLCSegmentName;
+import com.linkedin.pinot.controller.helix.core.realtime.SegmentCompletionManager;
+
+
+/**
+ * A resource class that fields the segmentCommit() message (and the upload of a realtime segment)
+ * from the servers. Servers go through {@link SegmentCompletionProtocol} to complete a segment and
+ * one of the servers commit the segment that has been completed.
+ */
+public class LLCSegmentCommit extends PinotSegmentUploadRestletResource {
+  private static Logger LOGGER = LoggerFactory.getLogger(LLCSegmentCommit.class);
+  long _offset;
+  String _segmentNameStr;
+  String _instanceId;
+
+  public LLCSegmentCommit() throws IOException {
+  }
+
+  @Override
+  @HttpVerb("post")
+  @Description("Uploads an LLC segment coming in from a server")
+  @Summary("Uploads an LLC segment coming in from a server")
+  @Paths({"/" + SegmentCompletionProtocol.MSG_TYPE_COMMMIT})
+  public Representation post(Representation entity) {
+    if (!extractParams()) {
+      return new StringRepresentation(SegmentCompletionProtocol.RESP_FAILED.toJsonString());
+    }
+    LOGGER.info("segment={} offset={} instance={} ", _segmentNameStr, _offset, _instanceId);
+    final SegmentCompletionManager segmentCompletionManager = getSegmentCompletionManager();
+
+    SegmentCompletionProtocol.Response response = segmentCompletionManager.segmentCommitStart(_segmentNameStr,
+        _instanceId, _offset);
+    if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
```

Comment:
Yes, there can be other responses, and they are being handled correctly (i.e. just return it to the caller).

I could move up the return comparing against not-equals-continue, but i wanted the same log, so decided to code it up like it is. Let me know if I can make this more obvious.